test/packages/integration_benchmarks/_dev/benchmark/rally/rally-benchmark.yml

```diff
@@ -0,0 +1,8 @@
+---
+description: Benchmark 20MiB of data ingested
+data_stream.name: test
```

Comment:
As this is a single value, does it mean the rally track will always be generated for a single dataset?

Reply:
the folder is per dataset, and the schemas are the same. so yes :)

Reply:
we could later figure out if there's the need to test "concurrent dataset ingestions" how to orchestrate them

Reply:
In the above, I assume `integration_benchmarks` is the package. It seems this is a `_dev` folder under the package level. When you say it is per dataset, is `rally` the dataset here? But then I don't get what there is a `data_stream.name`?

Reply:
@ruflin , yes, you are right, I mixed up the level of the `_dev` folder, sorry

to give more context you could reference @marc-gr's PR in `elastic-package`
https://github.com/elastic/elastic-package/pull/1232/files#diff-42b318802d9197a331edfa93fe80ac147ce9f76adbcc4aafa35a70930c3e20bdR59-R71

Like you will have from the spec already merged:
```
<package root>/
  _dev/
    benchmark/
      system/
        <scenario>.yml
```

You will have with my spec:
```
<package root>/
  _dev/
    benchmark/
      rally/
        <scenario>.yml
```

the content of every `<scenario>.yml` can vary, each of them targeting a specific `dataset` (imagine `system.cpu`). and potentially there could be multiple `<scenario>.yml` for the same `dataset` with different (well.. :)) scenarios

`rally` is not limited to a single dataset or even a single datastream, you can have a track with multiple of them, as far as I can see, but each corpus for each of them must be a separated file. specifically relating this to the corpus generator, the tool can generate one content for one single "dataset" (please, consider this as different schemas -A, B, C etc- for each dataset) at time


> But then I don't get what there is a `data_stream.name`?

in relation to the rally benchmark a `<scenario>.yml` will reference something like `data_stream.name: metrics-system.cpu-default`
it's the name of the data_stream. that's composed, according to the naming standard, by the dataset.

Reply:
So `rally` here is hardcoded and not related to package or dataset name. Inside is a list of scenarios. The part I stumbling over this that each scenario is for a single dataset: `data_stream.dataset`, so if the scenario would be inside, the config would not be needed. But having the scenario outside has the advantage, that a single scenario could be built for multiple dataset (which I think we should be able to eventually). Or is the idea here to run scenarios of scenarios?

I got confused by `data_stream.name`. First, that in your test scenario you call it `test` which is not a valid data stream naming scheme value. Second, `data_stream.name` doesn't really exist as part of the data stream naming scheme. I see this comes from the PR from marc. If we have it outside the package, shouldn't we specify `data_stream.dataset` here so we know which dataset it is for? I'll have a quick chat with @marc-gr also related to his PR, that might clarify things on my end.

Reply:
> The part I stumbling over this that each scenario is for a single dataset: `data_stream.dataset`, so if the scenario would be inside, the config would not be needed. But having the scenario outside has the advantage, that a single scenario could be built for multiple dataset (which I think we should be able to eventually). Or is the idea here to run scenarios of scenarios?

it's the same thing that confused me: indeed I'm not sure if a scenario could be built for multiple dataset, I don't see it feasible according to the current spec, and I honestly don't see a big advantage for that (it would be possible, but is it even useful? :)). so it could be the case that we can move the `rally`/`system` folders under `_dev` at dataset level.

I cannot guess why @marc-gr chose to use the package level, but I would probably have done the same, since that was already the case of the supported spec and it leads to less "folders pollution", and I would do only if the confusion you had as well cannot be avoided with better documentation :)

Reply:
Because of some historic renaming, `data_stream.name = data_stream.dataset`. We should rename it here to remove the confusion.

The discussion I had with @marc-gr is that for now we can keep it at the package level, we can always later also support it on the dataset level.

On the multi data set testing: I think there are use cases for this. Lets take k8s that has many dataset. I would want to have a rally track that ingests all dataset in parallel and not just one at the time. Just recently we have discovered that this can lead to a different behaviour in ES. I want the rally tracks to be as close to the real use case as possible.

Reply:
> Because of some historic renaming, `data_stream.name = data_stream.dataset`. We should rename it here to remove the confusion.

I'm not 100% sure about renaming in to `data_stream.dataset`
given `metrics-system.cpu-default`, `data_stream.dataset` should be `system.cpu`, while the "name of the data_stream" is indeed `metrics-system.cpu-default`

unless I missed something

Reply:
Are we putting the full name in? If yes, I kind of agree it is a different field. @marc-gr is this also the case on your end?